core/main/src/firebolt/firebolt_gateway.rs

```diff
@@ -191,11 +206,19 @@ impl FireboltGateway {
                 }
                 Err(e) => {
                     let deny_reason = e.reason;
-                    // return error for Api message
+
+                    TelemetryBuilder::stop_and_send_firebolt_metrics_timer(
+                        &platform_state.clone(),
+                        metrics_timer,
+                        format!("{}", deny_reason.get_rpc_error_code()),
```

Comment:
Ref

Reply:
suggestion: make a new method on deny_reason called like "get_reporting_error_code", so that each deny reason can have a unique error code. Unlike the rpc_error_code which uses the same error code for many deny reasons.